internal/providers/terraform/aws/dynamodb_table.go

```diff
@@ -21,41 +24,29 @@ func NewDynamoDBTable(d *schema.ResourceData, u *schema.ResourceData) *schema.Re
 	subResources := make([]*schema.Resource, 0)
 
 	billingMode := d.Get("billing_mode").String()
+	if billingMode != "PROVISIONED" {
+		log.Warnf("Skipping resource %s. Infracost currently does not support on-demand DynamoDB.", d.Address)
+		return nil
+	}
 
 	// Write capacity units (WCU)
-	if billingMode == "PROVISIONED" && d.Get("write_capacity").Exists() {
-		if billingMode != "PROVISIONED" {
-			log.Debugf("Skipping %s for %s. This attribute is only available for provisioned pricing.", "write_capacity", d.Address)
-		} else {
-			costComponents = append(costComponents, wcuCostComponent(d))
-		}
+	if billingMode == "PROVISIONED" {
+		costComponents = append(costComponents, wcuCostComponent(d))
 	}
 	// Read capacity units (RCU)
-	if billingMode == "PROVISIONED" && d.Get("read_capacity").Exists() {
-		if billingMode != "PROVISIONED" {
-			log.Debugf("Skipping %s for %s. This attribute is only available for provisioned pricing.", "read_capacity", d.Address)
-		} else {
-			costComponents = append(costComponents, rcuCostComponent(d))
-		}
+	if billingMode == "PROVISIONED" {
+		costComponents = append(costComponents, rcuCostComponent(d))
 	}
 
 	// Infracost usage data
 
 	// Write request units (WRU)
-	if u != nil && u.Get("monthly_write_request_units").Exists() {
-		if billingMode == "PROVISIONED" {
-			log.Debugf("Skipping %s usage data for %s. This usage data is only available for on-demand pricing.", "monthly_write_request_units", d.Address)
-		} else {
-			costComponents = append(costComponents, wruCostComponent(d, u))
-		}
+	if billingMode == "PAY_PER_REQUEST" {
+		costComponents = append(costComponents, wruCostComponent(d, u))
 	}
 	// Read request units (RRU)
-	if u != nil && u.Get("monthly_read_request_units").Exists() {
-		if billingMode == "PROVISIONED" {
-			log.Debugf("Skipping %s usage data for %s. This usage data is only available for on-demand pricing.", "monthly_read_request_units", d.Address)
-		} else {
-			costComponents = append(costComponents, rruCostComponent(d, u))
-		}
+	if billingMode == "PAY_PER_REQUEST" {
```

Comment:
Can we merge this with the if statement above as well?

Reply:
Done